src/main.rs

```diff
@@ -288,16 +299,23 @@ async fn main_inner() -> Result<(), anyhow::Error> {
 
     let mut threads = vec![];
     for i in 0..num_devices {
-        let c = config.clone();
-        let gpu = gpu_engine.clone();
-        let curr_stats_store = stats_store.clone();
-        threads.push(thread::spawn(move || {
-            run_thread(gpu, num_devices as u64, i as u32, c, benchmark, curr_stats_store)
-        }));
+        if devices_to_use.contains(&i) {
```

Comment:
I can't see where this variable is created. Does this compile?

Reply:
Indeed this var was removed while merging. Just fixed it. Compiled and tested with the Universe